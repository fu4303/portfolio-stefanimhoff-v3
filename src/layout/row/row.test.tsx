import React from 'react';

import { render } from '../../services/test-utils';

import { Row } from '.';

describe('Row', () => {
  test('renders correctly', () => {
    const { container } = render(<Row />);

    expect(container.firstChild).toMatchSnapshot();
  });

  describe('centered', () => {
    test('renders correctly', () => {
      const { container } = render(<Row variant="center" />);

      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('fullsize', () => {
    test('renders correctly', () => {
      const { container } = render(<Row variant="fullsize" />);

      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('only one item', () => {
    test('right renders correctly', () => {
      const { container } = render(<Row variant="rightOnly" />);

      expect(container.firstChild).toMatchSnapshot();
    });

    test('left renders correctly', () => {
      const { container } = render(<Row variant="leftOnly" />);

      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('with big item', () => {
    test('right renders correctly', () => {
      const { container } = render(<Row variant="bigRight" />);

      expect(container.firstChild).toMatchSnapshot();
    });

    test('left renders correctly', () => {
      const { container } = render(<Row variant="bigLeft" />);

      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('with two equal sized items', () => {
    test('renders correctly', () => {
      const { container } = render(<Row variant="equal" />);

      expect(container.firstChild).toMatchSnapshot();
    });

    test('aligned to start renders correctly', () => {
      const { container } = render(<Row variant="equal" vertical="start" />);

      expect(container.firstChild).toMatchSnapshot();
    });

    test('aligned to center renders correctly', () => {
      const { container } = render(<Row variant="equal" vertical="center" />);

      expect(container.firstChild).toMatchSnapshot();
    });

    test('aligned to end renders correctly', () => {
      const { container } = render(<Row variant="equal" vertical="end" />);

      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('with one item down', () => {
    test('right renders correctly', () => {
      const { container } = render(<Row variant="rightDown" />);

      expect(container.firstChild).toMatchSnapshot();
    });

    test('left renders correctly', () => {
      const { container } = render(<Row variant="leftDown" />);

      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('with variable amount of items', () => {
    test('renders correctly', () => {
      const { container } = render(<Row variant="variable" />);

      expect(container.firstChild).toMatchSnapshot();
    });

    test('aligned to start renders correctly', () => {
      const { container } = render(<Row variant="variable" vertical="start" />);

      expect(container.firstChild).toMatchSnapshot();
    });

    test('aligned to center renders correctly', () => {
      const { container } = render(
        <Row variant="variable" vertical="center" />
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    test('aligned to end renders correctly', () => {
      const { container } = render(<Row variant="variable" vertical="end" />);

      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
