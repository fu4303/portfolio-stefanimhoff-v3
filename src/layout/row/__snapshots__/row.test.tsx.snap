// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row centered renders correctly 1`] = `
.c0 {
  display: grid;
  grid-column: 1 / -1;
  grid-template-columns: repeat(18,1fr);
}

.c1 {
  display: grid;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  justify-items: center;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-box-pack: inherit;
  -webkit-justify-content: inherit;
  -ms-flex-pack: inherit;
  justify-content: inherit;
  justify-items: center;
  grid-column: 1 / -1;
}

<section
  class="c0"
>
  <div
    class="c1"
  />
</section>
`;

exports[`Row fullsize renders correctly 1`] = `
.c0 {
  display: grid;
  grid-column: 1 / -1;
  grid-template-columns: repeat(18,1fr);
}

.c1 {
  display: grid;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  justify-items: center;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-box-pack: inherit;
  -webkit-justify-content: inherit;
  -ms-flex-pack: inherit;
  justify-content: inherit;
  justify-items: inherit;
  grid-column: 1 / -1;
}

<section
  class="c0"
>
  <div
    class="c1"
  />
</section>
`;

exports[`Row only one item left renders correctly 1`] = `
.c0 {
  display: grid;
  grid-column: 1 / -1;
  grid-template-columns: repeat(18,1fr);
}

.c1 {
  display: grid;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  justify-items: center;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  gap: 5.55vw;
  grid-template-columns: 1fr;
  justify-items: center;
  grid-column: 1 / -1;
}

@media (min-width:768px) {
  .c1 {
    grid-template-columns: repeat(2,1fr);
  }
}

@media (min-width:768px) {
  .c1 > * {
    grid-column: 1;
  }
}

@media (min-width:768px) {
  .c1 {
    grid-column: 2 / -2;
  }
}

<section
  class="c0"
>
  <div
    class="c1"
  />
</section>
`;

exports[`Row only one item right renders correctly 1`] = `
.c0 {
  display: grid;
  grid-column: 1 / -1;
  grid-template-columns: repeat(18,1fr);
}

.c1 {
  display: grid;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  justify-items: center;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  gap: 5.55vw;
  grid-template-columns: 1fr;
  justify-items: center;
  grid-column: 1 / -1;
}

@media (min-width:768px) {
  .c1 {
    grid-template-columns: repeat(2,1fr);
  }
}

@media (min-width:768px) {
  .c1 > * {
    grid-column: 2;
  }
}

@media (min-width:768px) {
  .c1 {
    grid-column: 2 / -2;
  }
}

<section
  class="c0"
>
  <div
    class="c1"
  />
</section>
`;

exports[`Row renders correctly 1`] = `
.c0 {
  display: grid;
  grid-column: 1 / -1;
  grid-template-columns: repeat(18,1fr);
}

.c1 {
  display: grid;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  justify-items: center;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-box-pack: inherit;
  -webkit-justify-content: inherit;
  -ms-flex-pack: inherit;
  justify-content: inherit;
  justify-items: center;
  grid-column: 1 / -1;
}

<section
  class="c0"
>
  <div
    class="c1"
  />
</section>
`;

exports[`Row with big item left renders correctly 1`] = `
.c0 {
  display: grid;
  grid-column: 1 / -1;
  grid-template-columns: repeat(18,1fr);
}

.c1 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  justify-items: center;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  grid-column: 1 / -1;
}

@media (min-width:768px) {
  .c1 {
    grid-column: 1 / -6;
  }
}

<section
  class="c0"
>
  <div
    class="c1"
  />
</section>
`;

exports[`Row with big item right renders correctly 1`] = `
.c0 {
  display: grid;
  grid-column: 1 / -1;
  grid-template-columns: repeat(18,1fr);
}

.c1 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  justify-items: center;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  grid-column: 1 / -1;
}

@media (min-width:768px) {
  .c1 {
    grid-column: 6 / -1;
  }
}

<section
  class="c0"
>
  <div
    class="c1"
  />
</section>
`;

exports[`Row with one item down left renders correctly 1`] = `
.c0 {
  display: grid;
  grid-column: 1 / -1;
  grid-template-columns: repeat(18,1fr);
}

.c1 {
  display: grid;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  justify-items: center;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  gap: 5.55vw;
  grid-template-columns: 1fr;
  justify-items: center;
  grid-column: 1 / -1;
}

@media (min-width:768px) {
  .c1 {
    grid-template-columns: repeat(2,1fr);
  }
}

@media (min-width:768px) {
  .c1 > *:first-child {
    margin-top: 11vw;
  }
}

@media (min-width:768px) {
  .c1 {
    grid-column: 2 / -2;
  }
}

<section
  class="c0"
>
  <div
    class="c1"
  />
</section>
`;

exports[`Row with one item down right renders correctly 1`] = `
.c0 {
  display: grid;
  grid-column: 1 / -1;
  grid-template-columns: repeat(18,1fr);
}

.c1 {
  display: grid;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  justify-items: center;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  gap: 5.55vw;
  grid-template-columns: 1fr;
  justify-items: center;
  grid-column: 1 / -1;
}

@media (min-width:768px) {
  .c1 {
    grid-template-columns: repeat(2,1fr);
  }
}

@media (min-width:768px) {
  .c1 > *:last-child {
    margin-top: 11vw;
  }
}

@media (min-width:768px) {
  .c1 {
    grid-column: 2 / -2;
  }
}

<section
  class="c0"
>
  <div
    class="c1"
  />
</section>
`;

exports[`Row with two equal sized items aligned to center renders correctly 1`] = `
.c0 {
  display: grid;
  grid-column: 1 / -1;
  grid-template-columns: repeat(18,1fr);
}

.c1 {
  display: grid;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  justify-items: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 5.55vw;
  grid-template-columns: 1fr;
  justify-items: center;
  grid-column: 1 / -1;
}

@media (min-width:768px) {
  .c1 {
    grid-template-columns: repeat(2,1fr);
  }
}

@media (min-width:768px) {
  .c1 {
    grid-column: 2 / -2;
  }
}

<section
  class="c0"
>
  <div
    class="c1"
  />
</section>
`;

exports[`Row with two equal sized items aligned to end renders correctly 1`] = `
.c0 {
  display: grid;
  grid-column: 1 / -1;
  grid-template-columns: repeat(18,1fr);
}

.c1 {
  display: grid;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  justify-items: center;
  -webkit-align-items: end;
  -webkit-box-align: end;
  -ms-flex-align: end;
  align-items: end;
  gap: 5.55vw;
  grid-template-columns: 1fr;
  justify-items: center;
  grid-column: 1 / -1;
}

@media (min-width:768px) {
  .c1 {
    grid-template-columns: repeat(2,1fr);
  }
}

@media (min-width:768px) {
  .c1 {
    grid-column: 2 / -2;
  }
}

<section
  class="c0"
>
  <div
    class="c1"
  />
</section>
`;

exports[`Row with two equal sized items aligned to start renders correctly 1`] = `
.c0 {
  display: grid;
  grid-column: 1 / -1;
  grid-template-columns: repeat(18,1fr);
}

.c1 {
  display: grid;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  justify-items: center;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  gap: 5.55vw;
  grid-template-columns: 1fr;
  justify-items: center;
  grid-column: 1 / -1;
}

@media (min-width:768px) {
  .c1 {
    grid-template-columns: repeat(2,1fr);
  }
}

@media (min-width:768px) {
  .c1 {
    grid-column: 2 / -2;
  }
}

<section
  class="c0"
>
  <div
    class="c1"
  />
</section>
`;

exports[`Row with two equal sized items renders correctly 1`] = `
.c0 {
  display: grid;
  grid-column: 1 / -1;
  grid-template-columns: repeat(18,1fr);
}

.c1 {
  display: grid;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  justify-items: center;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  gap: 5.55vw;
  grid-template-columns: 1fr;
  justify-items: center;
  grid-column: 1 / -1;
}

@media (min-width:768px) {
  .c1 {
    grid-template-columns: repeat(2,1fr);
  }
}

@media (min-width:768px) {
  .c1 {
    grid-column: 2 / -2;
  }
}

<section
  class="c0"
>
  <div
    class="c1"
  />
</section>
`;

exports[`Row with variable amount of items aligned to center renders correctly 1`] = `
.c0 {
  display: grid;
  grid-column: 1 / -1;
  grid-template-columns: repeat(18,1fr);
}

.c1 {
  display: grid;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  justify-items: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: clamp(2.488rem,5.32vw,2.986rem);
  grid-column: 1 / -1;
  grid-template-columns: repeat(auto-fit,minmax(15rem,1fr));
}

<section
  class="c0"
>
  <div
    class="c1"
  />
</section>
`;

exports[`Row with variable amount of items aligned to end renders correctly 1`] = `
.c0 {
  display: grid;
  grid-column: 1 / -1;
  grid-template-columns: repeat(18,1fr);
}

.c1 {
  display: grid;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  justify-items: center;
  -webkit-align-items: end;
  -webkit-box-align: end;
  -ms-flex-align: end;
  align-items: end;
  gap: clamp(2.488rem,5.32vw,2.986rem);
  grid-column: 1 / -1;
  grid-template-columns: repeat(auto-fit,minmax(15rem,1fr));
}

<section
  class="c0"
>
  <div
    class="c1"
  />
</section>
`;

exports[`Row with variable amount of items aligned to start renders correctly 1`] = `
.c0 {
  display: grid;
  grid-column: 1 / -1;
  grid-template-columns: repeat(18,1fr);
}

.c1 {
  display: grid;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  justify-items: center;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  gap: clamp(2.488rem,5.32vw,2.986rem);
  grid-column: 1 / -1;
  grid-template-columns: repeat(auto-fit,minmax(15rem,1fr));
}

<section
  class="c0"
>
  <div
    class="c1"
  />
</section>
`;

exports[`Row with variable amount of items renders correctly 1`] = `
.c0 {
  display: grid;
  grid-column: 1 / -1;
  grid-template-columns: repeat(18,1fr);
}

.c1 {
  display: grid;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  justify-items: center;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  gap: clamp(2.488rem,5.32vw,2.986rem);
  grid-column: 1 / -1;
  grid-template-columns: repeat(auto-fit,minmax(15rem,1fr));
}

<section
  class="c0"
>
  <div
    class="c1"
  />
</section>
`;
